modules/core/05-port/types/module.go

```diff
@@ -82,4 +82,11 @@ type IBCModule interface {
 		packet channeltypes.Packet,
 		relayer sdk.AccAddress,
 	) error
+
+	NegotiateAppVersion(
+		ctx sdk.Context,
+		portID string,
+		counterparty channeltypes.Counterparty,
+		counterpartyVersion string,
```

Comment:
Information not provided:
- channel id
- channel ordering
- channel state
- connection hop (connection channel is attached to)

channel id cannot necessarily be known by OnChanOpenTry. Channel state shouldn't matter

Should we include channel ordering and connection hops? We could always break API later to add it? I have no strong opinion except knowing the associated connection may be useful

Reply:
I think there is no harm in adding it now. If I'm understanding you correctly, no current module requires these parameters for negotiating the counterparty version but adding them now will let future modules use as required?

Reply:
Correct. To the best of my knowledge, interchain accounts is the first module introducing multiple valid versions. It's hard to know what version negotiation will look like for applications, but I agree probably best to be as flexible as possible